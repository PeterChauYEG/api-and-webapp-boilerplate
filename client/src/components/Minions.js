import React, { Component } from 'react'

// import components
import HoverButton from './HoverButton'
import MinionsContainer from './MinionsContainer'
import MinionForm from './MinionForm'

// import css
import '../CSS/Minions.css'

class Minions extends Component {
  renderForm(actions, user, formIsOpen) {
    return formIsOpen ?
      <MinionForm
        actions={actions}
        user={user} /> :
        null
  }

  render() {
    const {
      actions,
      minions,
      user,
    } = this.props;

    return (
      <div className="Minions pa4">
        <h2>Minions</h2>
        <MinionsContainer minions={minions.minions} />

        {
          /* render form */
          this.renderForm(actions, user, minions.formIsOpen)
        }

        <HoverButton
          action={actions.toggleMinionForm}
          backgroundColor={minions.formIsOpen ? 'bg-near-white' : 'bg-light-blue'}
          icon={minions.formIsOpen ? 'fa-close' : 'fa-plus' } />

      </div>
    );
  }
}

Minions.propTypes = {
  actions: React.PropTypes.object.isRequired,
  minions: React.PropTypes.object.isRequired,
}

export default Minions;
