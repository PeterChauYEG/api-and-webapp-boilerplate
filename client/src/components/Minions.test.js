import React from 'react';

// import test dependencies 
import { shallow } from 'enzyme';

// import component
import Minions from './Minions'

it('renders without crashing', () => {
  const actions = {}
  const minions = {}
  
  shallow(
    <Minions
      actions={actions}
      minions={minions}
      />
  );
});