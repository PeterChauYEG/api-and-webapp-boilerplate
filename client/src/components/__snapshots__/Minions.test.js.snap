exports[`test renders without crashing 1`] = `
<div
  className="Minions pa4">
  <h1>
    Minions
  </h1>
  <div>
    <p>
      There\'s no minions yet. You should add one!
    </p>
  </div>
  <div
    className="HoverButton bg-light-blue"
    onClick={[Function]}>
    <i
      className="fa fa-plus fa-2x" />
  </div>
</div>
`;
